### ISSUE TITLE
Consider adding JSON/key value pair  output to docker-env command

### ISSUE THREAD

**Issue by tcsmith** [kubernetes/minikube#12242]
We use python to do a good amount of automation at my shop. When iterating locally it would be nice to be able to have a cross platform/language agnostic output of the `docker-env `command that a tool written in a non-shell script language could use. 

```
$ minikube docker-env
You can further specify your shell with either 'cmd' or 'powershell' with the --shell flag.

SET DOCKER_TLS_VERIFY=1
SET DOCKER_HOST=tcp://127.0.0.1:61041
SET DOCKER_CERT_PATH=C:\Users\user\.minikube\certs
SET MINIKUBE_ACTIVE_DOCKERD=minikube
REM To point your shell to minikube's docker-daemon, run:
REM @FOR /f "tokens=*" %i IN ('minikube -p minikube docker-env') DO @%i
```
Ideally something like this:

```
$ minikube docker-env --output-kvp
{ 
"DOCKER_TLS_VERIFY": 1,
"DOCKER_HOST":"tcp://127.0.0.1:61041",
"DOCKER_CERT_PATH":"C:\\Users\\user\\.minikube\\certs",
"MINIKUBE_ACTIVE_DOCKERD":"minikube"
}
```

I realize this could be done with regex libraries in the language of choice but having a convenience function would be fantastic.

**Comment by medyagh** [comment]
@tcsmith that sounds like a great idea !

We already have an option for docker-env to choose the shell type
One option is shell=none

I would review any pr that would add Jason feature to this command.

Would you like to contribute this feature?

**Comment by tcsmith** [comment]
Ahhhhh, I knew of the `--shell` option but I did not know about the none option. the help string doesnt mention that and I, lazily, did not dig thru the source XD.

`--shell=none` definitely makes it much easier to parse! 

i would def be interested in adding JSON/KVP output though. what should the option be? we could have a shell type of `json `or `kvp `but that doesn't seem 'right'. on the other hand, the option i suggested, `--output-kvp`, seems a tad bit bloated?

**Comment by afbjorklund** [comment]
Apparently it was added already in a02541d18bf6fb9110e6b8efab6f76fc27fa6514 but not mentioned in the docs. (and not in the upstream `docker-machine env`)

The other commands take a `--output` option, so you can do the same here. And have options like text (default), json, yaml

`minikube docker-env --shell none --output json`

`minikube podman-env --shell none --output yaml`

**Comment by afbjorklund** [comment]
The shell "framework" itself needs some extension, to add a start of the list and the end of the list.

You know the { } crud that is the JSON format, and that makes it so unsuitable for streaming ? 😀

Currrently it only has `Prefix` and `Suffix`, but those relate to the line and not to the document.

```
{{ .Prefix }}%s{{ .Delimiter }}{{ .DockerTLSVerify }}{{ .Suffix }}
{{ .Prefix }}%s{{ .Delimiter }}{{ .DockerHost }}{{ .Suffix }}
{{ .Prefix }}%s{{ .Delimiter }}{{ .DockerCertPath }}{{ .Suffix }}
```

----

Empty file ? Illegal JSON

Append a line ? Not in JSON.

For the rest, see https://jsonlines.org/

And of course also https://hjson.github.io/

**Comment by afbjorklund** [comment]
I guess the most likely is to just dump the shell template altogether, and just dump the vars.

Could do the same for the regular text output, the current "none" shell is a bit of a hack...

**Comment by afbjorklund** [comment]
Is KVP actually a thing ? Like, a real format

**Comment by tcsmith** [comment]
Fantastic, and thank you.

I was able to use the --shell=none output to easily parse it, that output is basically KVP, in my python tools. Your change will even make it easer!

I realized after the fact that K8S/minkube are written in go. I don't have any experience with Go and given the phase my project is in, I wouldnt have the time to learn. Once my project ships, I will learn Go and come back to contribute. I'm learning and loving K8S and just found out about Minikube and its a gamechanger being able to iterate locally, and this feature will help immensely.

Thanks again!
Todd

👁❤🧊🤗
